server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -144,6 +145,16 @@ public RestController(
         );
     }
 
+    /**
+     * Returns an iterator over registered REST method handlers.
+     * @return {@link Iterator} of {@link MethodHandlers}
+     */
+    public Iterator<MethodHandlers> getAllHandlers() {
+        List<MethodHandlers> methodHandlers = new ArrayList<>();
+        handlers.retrieveAll().forEachRemaining(methodHandlers::add);
+        return methodHandlers.iterator();
```

Comment:
No necessarily better, but more functional:

```
return StreamSupport
            .stream(Spliterators.spliteratorUnknownSize(handlers.retrieveAll(), Spliterator.ORDERED), false)
            .map(MethodHandlers.class::cast)
            .iterator();
```

Reply:
I hate it ;)

Reply:
I know I know :)